src/managers/ApplicationCommandManager.js

```diff
@@ -177,6 +199,139 @@ class ApplicationCommandManager extends BaseManager {
       default_permission: command.defaultPermission,
     };
   }
+
+  /**
+   * Fetches the permissions for one or multiple commands.
+   * <warn>When calling this on ApplicationCommandManager, guildID is required.
+   * To fetch all permissions for an uncached guild use `fetchPermissions(undefined, '123456789012345678')`</warn>
```

Comment:
I just wanted to make a note here about passing `undefined`. I don't love this pattern, but as Monbrey said in the discord, I don't love having to pass { command: '123456789012345678' } either.

Reply:
If the guild ID is the required one, put it first?

Reply:
when called from a GuildApplicationCommandManager its not required, although I suppose I could just use super in that class.

Reply:
Okay, I tried to do this, and it simply doesn't work as wanted.
Putting guildID as the first parameter means implementing `fetchPermissions` and `setPermissions` in `GuildApplicationCommandManager` which we can do in js, but because guildID is not a parameter there, typescript really doesn't like it, suggesting this is a bad pattern in general.

Reply:
Any other suggestions for this (or a suggestoin for how to make typescript happy with different method signatures)?

Reply:
`override` still doesn't let you remove / repurpose parameters in the signature though from what I can tell.

Reply:
It's not nice, but you could allow either just a command resolvable, or an object with it and the guild.

Reply:
There is precedent for this elsewhere in the library actulaly, so it works, will do.

Reply:
A CommandResolvable can't be accepted as the only parameter because this.resolve (when providing an id) can't resolve guild application commands, which are required when not passing guildID so that we can get command.guild.id instead. And typescript will complain if I try to make it ApplicationCommand | { guildID: Snowflake | command?: ApplciationCommandResolvable } and then override that to ApplicationCommandResolvable in the subclass

Reply:
Excuse a stranger, but if I've understood correctly the goal here is to somehow put the `guildId` parameter to the `fetchPermissions` method (possibly to more methods) and at the same time satisfy the TypeScript compiler in the `GuildApplicationCommandManager`, right?

So I've created a small "demo" of my approach on the TS Playground [here](https://www.typescriptlang.org/play?#code/PTAEGUBcFcCNQIIAckBsCWBjAhpdB7AOwGF8BbM7QgE1E1WwGdGAoepxxFDHPI0ilVoBvFqHGgkcHqEgBTRpABcoRQCd0hAOYBuFmImYi66Jkj41ACgCUoURIeyAFukYA6eYtABeUACI-PQcAXxZQlkgATyQ5CEJ8AHcAMwYAa1jfdU1dfSiYrjQsXAISckoaACUFfFQAN2xYVAyCnmL+MqFQAB84xJTsdL082IBxaHRUamRC3hKBcuoqxhr6xubhUC1xyYBJahVweOS0uQAaOg6aFWnWvlLBSuq6hqbQYKHo2IB5JDvsVAAPAAVc4AaVAcgAHvIaJx0pF8ElQECAHw+UAABSwqQBGOwajw-2BKLBaIAZKAvmR0JBgaS9GwGMwWkU7vMhABZKjYLRyNR2AziKSNLCgJJySCYJwYvnU5glRiWLSofCwf5fNRjCbUdk0AD8KkFjhurLml0WT1Wrx6RocWsmJtm7QeFuWzzW3VAtokPz+gPtU24pudCyWKxeZ38RhdfhRRusKlq+HQIm94hAoGo+FU5AlLm0oCcfLkRtC4XYzIDjra9wWXMIPL5EOhclhLKdtc53N5-PsEmFMnFkulstcjAVlmjCwN7Zrutd4bWCdASZTAscEgzWZzZDz2ULxdLYX0RkIXmVqv+9cb-N8hDkCVnbPN157NgZF7VqFffLcQ6lMpqHK47GO+LCfle3a-v+I5AWOE5+AAjAATAAzH41h6BB35QWof4SgBo7yqB96PtWz4ujYmHgSqX4-nhMGAcBE4bFs2p7CoSFoX45xTkInEoehbzUdh9H4cOTHwaBrHbNQHH+IJPEXC6KikU+ZqUbYwQibRkENj24mEXBxFnpYMnsfsCnccJDKnuesn0eialVkGHbzvRYEsBmADqFipOBDm4YZsHMaBImBfp0EESFUmmVx6HhdqYmMURIGmWp5EaQsVEMhmAAiRCQKACR+QFSVBSlxlpYq5m7JZ8VKXxVxWUJ2lYRFN7BZJJk1ZssnyQ1vHmqpD7qSGQg2DZZWTMl0XddVZl9RZAnWW1QA) and with this solution you can pass all combinations: **no params**, **only command** and **guild id + previous combinations**. I've based it on the @SpaceEEC suggestion to put either a resolvable or an object with `guildId` and the command.

(I know that I wrote it entirely in TypeScript, and it may or may not be 100% applicable but maybe at the very least it will notch you guys in the right direction, hope I've helped)